examples/dao/test/failing-path.js

```diff
@@ -0,0 +1,658 @@
+const { AccountStore } = require('@algo-builder/runtime');
+const { types } = require('@algo-builder/web');
+const { assert } = require('chai');
+const { Context, initialBalance, minSupport, deposit } = require('./common');
+const { ProposalType, Vote } = require('../scripts/run/common/common');
+const { mkProposalTx, mkDepositVoteTokenTx, votingStart, votingEnd, executeBefore, mkWithdrawVoteDepositTx, mkClearVoteRecordTx, mkClearProposalTx } = require('../scripts/run/common/tx-params');
+
+const now = Math.round(new Date().getTime() / 1000);
+
+const RUNTIME_ERR1009 = 'RUNTIME_ERR1009: TEAL runtime encountered err opcode';
+const INDEX_OUT_OF_BOUND_ERR = 'RUNTIME_ERR1008: Index out of bound';
+const INTEGER_UNDERFLOW_ERR = 'Result of current operation caused integer underflow';
+const APP_NOT_FOUND = 'RUNTIME_ERR1306: Application Index 1 not found or is invalid';
+
+describe('DAO - Failing Paths', function () {
+  let master, creator, proposer, voterA, voterB;
+  let depositLsigAcc, daoFundLsigAcc, proposalLsigAcc;
+  let ctx;
+
+  function setUpCtx () {
+    master = new AccountStore(1000e6);
+    creator = new AccountStore(initialBalance);
+    proposer = new AccountStore(initialBalance);
+    voterA = new AccountStore(initialBalance);
+    voterB = new AccountStore(initialBalance);
+    depositLsigAcc = new AccountStore(initialBalance); // runtime.account of depositLsig.address()
+    daoFundLsigAcc = new AccountStore(initialBalance);
+    proposalLsigAcc = new AccountStore(initialBalance);
+
+    ctx = new Context(
+      master, creator, proposer, voterA, voterB,
+      depositLsigAcc, daoFundLsigAcc, proposalLsigAcc
+    );
+  }
+
```

Comment:
could you please add a test verifying that only creator can update `deposit_lsig` account ?

Reply:
yes